### DISCUSSION TITLE
Welcome to Hoppscotch Discussions!

### DISCUSSION THREAD

**Discussion by liyasthomas** [hoppscotch/hoppscotch#1368] [General]
## 👋 &nbsp; Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and be open-minded. Remember that this is a community we build together.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Itshivamgithub** [comment]
how can i work localhost APIs

**Reply by liyasthomas** [reply]
This comment might help: https://github.com/hoppscotch/hoppscotch/discussions/2051#discussioncomment-1887545

Also, read the Hoppscotch Setup documentation: https://docs.hoppscotch.io/documentation/getting-started/setup#locally-served-apis

**Comment by Subhan-Barakzai-385** [comment]
How to access Hoppcosts in Localhost

**Reply by liyasthomas** [reply]
This comment might help: https://github.com/hoppscotch/hoppscotch/discussions/2051#discussioncomment-1887545

Also, read the Hoppscotch Setup documentation: https://docs.hoppscotch.io/documentation/getting-started/setup#locally-served-apis

**Comment by outradriss** [comment]
Hello,

I tried to run the image locally but I encountered this error:


=> ERROR [fe_builder 2/2] RUN pnpm run generate                                                                                                                                             45.8s 
 => CANCELED [sh_admin_builder 2/3] RUN pnpm run build --outDir dist-multiport-setup                                                                                                         46.2s 
 => [backend 2/6] WORKDIR /usr/src/app/packages/hoppscotch-backend                                                                                                                            0.1s 
 => [backend 3/6] RUN pnpm exec prisma generate                                                                                                                                              10.4s 
 => CANCELED [backend 4/6] RUN pnpm run build                                                                                                                                                24.8s 
------
 > [fe_builder 2/2] RUN pnpm run generate:
3.522
3.522 > @hoppscotch/selfhost-web@2024.7.0 generate /usr/src/app/packages/hoppscotch-selfhost-web
3.522 > pnpm run build
3.522
7.431
7.431 > @hoppscotch/selfhost-web@2024.7.0 build /usr/src/app/packages/hoppscotch-selfhost-web
7.431 > node --max_old_space_size=4096 ./node_modules/vite/bin/vite.js build
7.431
12.80 vite v4.5.0 building for production...
14.05 transforming...
45.34
45.34 import-meta-env v0.4.10
45.34 ✓ environment files are generated.
45.34 Remember to inject (`npx import-meta-env`) environment variables before serving your application.
45.34
45.35 ✓ 987 modules transformed.
45.35 [vite-plugin-pwa:build] Could not load /@vite-plugin-pages/generated-pages?id=virtual:generated-pages (imported by ../hoppscotch-common/src/modules/router.ts): Cannot read properties of undefined (reading 'endsWith')
45.35 error during build:
45.35 TypeError: Could not load /@vite-plugin-pages/generated-pages?id=virtual:generated-pages (imported by ../hoppscotch-common/src/modules/router.ts): Cannot read properties of undefined (reading 'endsWith')
45.35     at removeMaybeSuffix (/usr/src/app/node_modules/.pnpm/vite-plugin-pages-sitemap@1.6.1/node_modules/vite-plugin-pages-sitemap/dist/index.js:4189:12)
45.35     at /usr/src/app/node_modules/.pnpm/vite-plugin-pages-sitemap@1.6.1/node_modules/vite-plugin-pages-sitemap/dist/index.js:4198:13
45.35     at Array.map (<anonymous>)
45.35     at getSitemapLinks (/usr/src/app/node_modules/.pnpm/vite-plugin-pages-sitemap@1.6.1/node_modules/vite-plugin-pages-sitemap/dist/index.js:4197:276)
45.35     at generateSitemap (/usr/src/app/node_modules/.pnpm/vite-plugin-pages-sitemap@1.6.1/node_modules/vite-plugin-pages-sitemap/dist/index.js:4260:3)
45.35     at Object.onRoutesGenerated (file:///usr/src/app/packages/hoppscotch-selfhost-web/vite.config.ts.timestamp-1722961977343-9a0de72ddbdb2.mjs:221:9)
45.35     at computeVueRoutes (file:///usr/src/app/node_modules/.pnpm/vite-plugin-pages@0.31.0_@vue+compiler-sfc@3.3.10_vite@4.5.0_@types+node@18.18.8_sass@1.69.5_terser@5.31.0_/node_modules/vite-plugin-pages/dist/index.mjs:543:90)
45.35     at resolveVueRoutes (file:///usr/src/app/node_modules/.pnpm/vite-plugin-pages@0.31.0_@vue+compiler-sfc@3.3.10_vite@4.5.0_@types+node@18.18.8_sass@1.69.5_terser@5.31.0_/node_modules/vite-plugin-pages/dist/index.mjs:548:29)
45.35     at Object.resolveRoutes (file:///usr/src/app/node_modules/.pnpm/vite-plugin-pages@0.31.0_@vue+compiler-sfc@3.3.10_vite@4.5.0_@types+node@18.18.8_sass@1.69.5_terser@5.31.0_/node_modules/vite-plugin-pages/dist/index.mjs:587:14)
45.35     at PageContext.resolveRoutes (file:///usr/src/app/node_modules/.pnpm/vite-plugin-pages@0.31.0_@vue+compiler-sfc@3.3.10_vite@4.5.0_@types+node@18.18.8_sass@1.69.5_terser@5.31.0_/node_modules/vite-plugin-pages/dist/index.mjs:936:34)
45.61  ELIFECYCLE  Command failed with exit code 1.
45.66  ELIFECYCLE  Command failed with exit code 1.
------

 8 warnings found (use --debug to expand):
 - FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 1)
 - FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 18)
 - FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 34)
 - FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 38)
 - FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 49)
 - FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 55)
 - FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 68)
 - UndefinedVar: Usage of undefined variable '$DATABASE_URL' (line 29)
prod.Dockerfile:36
--------------------
  34 |     FROM base_builder as fe_builder
  35 |     WORKDIR /usr/src/app/packages/hoppscotch-selfhost-web
  36 | >>> RUN pnpm run generate
  37 |
  38 |     FROM caddy:2-alpine as app
--------------------
ERROR: failed to solve: process "/bin/sh -c pnpm run generate" did not complete successfully: exit code: 1

 *  The terminal process "C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -Command docker build --pull --rm -f "prod.Dockerfile" -t hoppscotch:latest "."" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it. 


Do you have any ideas, please?
